### PR TITLE
Make puma log silence completely

### PR DESCRIPTION
**Pull request by benoittgt**
Continue https://github.com/rspec/rspec-rails/pull/2011

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -197,12 +198,13 @@ namespace :no_active_record do
           sh "bundle binstubs railties" unless File.exist?("bin/rails")
 
           application_file = File.read("config/application.rb")
+          application_file.gsub!("config.assets.enabled = true", "config.assets.enabled = false")
+          application_file.gsub!('# require "sprockets/railtie"', 'require "sprockets/railtie"')
```
Comment (benoittgt) [inline Rakefile]: I felt quite dirty to do this.

File: Changelog.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ Enhancements:
 * The scaffold generator now generates request specs in preference to controller specs.
   (Luka Lüdicke, #2288)
 * Add configuration option to disable ActiveRecord. (Jon Rowe, Phil Pirozhkov, Hermann Mayer, #2266)
+* Remove Puma log when running system specs. (ta1kt0me, Benoit Tigeot, #2011)
```
Comment (benoittgt) [inline Changelog.md]: Should I reference this one?

Comment (JonRowe) [reply Changelog.md]: Yes

Comment (JonRowe) [inline Changelog.md]:
```suggestion
*  Set `ActionDispatch::SystemTesting::Server.silence_puma = true` when running system specs.
    (ta1kt0me, Benoit Tigeot, #2289)
```

**Pull request merged**